### ISSUE TITLE
Undo support for delete waypoint

### ISSUE THREAD

**Issue by Matze2** [cgeo/cgeo#7096]
Recently I managed again to delete a waypoint (with a prepared calculator) by accident.
Therefore I tried to implement the Undo capability for the Delete action.

This works but since the Undo action triggers a Re-Insert of the waypoint (and waypoints are only ordered by their id) it will be moved to the end of the waypoint list which is not an ideal solution.

To make it better waypoints would need support for free ordering.

The question is if there is interest in a semi-optimal Undo implementation where an unwanted waypoint deletion is restored but positioned at the end of the list. Then I would prepare a PR.

**Comment by SammysHP** [comment]
Undo should do what it says: undo an action and restore the state as if nothing has happened.

**Comment by Matze2** [comment]
Thanks for clarification.

Without the possibility to freely order the waypoint list (which would be also a nice feature BTW) this can't be done. Reordering would involve database changes and with all its implications like compatibility this is much more complex than converting a few code lines into a command to realize an Undo.

**Comment by pstorch** [comment]
Sorting of waypoints looks like #2677

**Comment by Matze2** [comment]
Yes somehow. But #2677 talks about sorting is based on existing waypoint attributes (in detail: waypoint name).
To manually influence the order waypoints are displayed a "position" attributes is necessary as a prerequisite.

**Comment by pstorch** [comment]
OK, but the discussion fits better there instead of here (discussing undo).

**Comment by Lineflyer** [comment]
Back to the original issue:
We do already have a confirmation dialog before deleting a waypoint. If someone prefers an Undo operation, it should replace the confirmation dialog.

IMHO I would prefer to confirmation dialog instead of an Undo, because a seldomly use the Undo and mostly I only recognize an accidental deletion only after the Undo bar has already vanished.

**Comment by Matze2** [comment]
My c:geo from last night deletes waypoints without confirmation,

**Comment by Lineflyer** [comment]
Oh, you are right. I mixed it up with the discarding of waypoint changes. The temperatures :(

**Comment by Matze2** [comment]
The point is that a lot of actions in cgeo got Undo support on actions that modify data instead of confirmation dialogs - also within the waypoint context menu. Shouldn't there be a general strategy how we deal with this?

I like the Undo toasts but I agree it would be nice if there is more time to click on it.

Based on the feedback in #2677 it is clear that the necessary infrastructure for Undo is not available and there is also no interest to have it implemented. Therefore the options "confirmation dialog" or "reject this issue" remain.

**Comment by Lineflyer** [comment]
I vote for confirmation dialog. I rarely delete any waypoint and I assume this might apply to many users, so it wont bother to much.
